src/lib/broadcast.jl

```diff
@@ -237,28 +231,33 @@ function dual_function(f::F) where F
 end
 
 @inline function broadcast_forward(f, args::Vararg{Any,N}) where N
-  T = Broadcast.combine_eltypes(f, args)
+  valN = Val(N)
   out = dual_function(f).(args...)
   eltype(out) <: Dual || return (out, _ -> nothing)
```

Comment:
`broadcast_forward` is mostly edited for clarity. The exception is that this shortcut `return (out, _ -> nothing)` didn't actually work before, as the caller would do what the line `(nothing, nothing, dargs...)` now does here, which isn't desired in this case.